packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts

```diff
@@ -66,18 +69,63 @@ export class DLCBTCPorTransport extends SubscriptionTransport<TransportTypes> {
         },
       }
     }
-    await this.responseCache.write(this.name, [{ params: [], response }])
+    await this.responseCache.write(this.name, [{ params: param, response }])
   }
 
-  async _handleRequest(): Promise<AdapterResponse<TransportTypes['Response']>> {
+  async _handleRequest(param: RequestParams): Promise<AdapterResponse<TransportTypes['Response']>> {
+    const { network } = param
+    let networkName = 'DEFAULT'
+
+    let networkEnvName = `RPC_URL`
+    let chainIdEnvName = `CHAIN_ID`
+    let dlcContractEnvName = `DLC_CONTRACT`
+
+    let rpcUrl = this.settings.RPC_URL
+    let chainId = this.settings.CHAIN_ID
+    let dlcContractAddress = this.settings.DLC_CONTRACT
+
+    // If network is provided, use the network specific environment variables. We still support the old environment variables for backward compatibility.
+    if (network) {
+      networkName = network.toUpperCase()
+      networkEnvName = `${networkName}_RPC_URL`
+      chainIdEnvName = `${networkName}_CHAIN_ID`
+      dlcContractEnvName = `${networkName}_DLC_CONTRACT`
+
+      rpcUrl = process.env[networkEnvName] as string
+      chainId = Number(process.env[chainIdEnvName])
+      dlcContractAddress = process.env[dlcContractEnvName] as string
+    }
+
+    if (!rpcUrl || !dlcContractAddress || !chainId || isNaN(chainId)) {
+      throw new AdapterInputError({
+        statusCode: 400,
+        message: `Missing '${networkEnvName}', '${chainIdEnvName}' or '${dlcContractEnvName}' environment variables.`,
+      })
+    }
+
+    if (!network) {
+      logger.warn(
+        `Network is not provided. Reading from RPC_URL, CHAIN_ID and DLC_CONTRACT environment variables. Please use the network parameter to specify the network in the environment variables and in the request.`,
+      )
+    }
+
+    if (!this.providers[networkName]) {
+      this.providers[networkName] = new ethers.providers.JsonRpcProvider(rpcUrl, chainId)
```

Comment:
maybe a silly question, but is there any lag on the first request to a new chain because we have to set up this object?

Reply:
There will be a lag on the very first request because it needs an active subscription set to execute the code. On a new network request whether there will be a lag completely depends on the timing. The object setup is synchronous and super fast, but each background cycle runs every 10s, so if a new requests comes right before a new background cycle has started, a response will be immediate, otherwise worst case scenario there can be a lag of maximum 10s

Reply:
It will just try again in every background execution loop